napari/components/camera.py

```diff
@@ -26,16 +26,19 @@ class Camera(EventedModel):
         sets of Euler angles may lead to the same view.
     perspective : float
         Perspective (aka "field of view" in vispy) of the camera (if 3D).
-    interactive : bool
-        If the camera interactivity is enabled or not.
+    mouse_pan : bool
+        If the camera interactive panning with the mouse is enabled or not.
+    mouse_zoom : bool
+        If the camera interactive zooming with the mouse is enabled or not.
     """
 
     # fields
     center: Tuple[float, float, float] = (0.0, 0.0, 0.0)
     zoom: float = 1.0
     angles: Tuple[float, float, float] = (0.0, 0.0, 90.0)
     perspective: float = 0
-    interactive: bool = True
+    mouse_pan: bool = True
+    mouse_zoom: bool = True
```

Comment:
There should be a backward compatibility layer with a deprecation warning. 

like this:
```python

@property
def interactive(self):
    warnings.warn("`Camera.interactive` is deprecated since 0.5.0 and will be removed in 0.6.0", category=FutureWarning)
    return self.mouse_pan and self.mouse_zoom

@interactive.setter
def interactive(self, value):
    warnings.warn("`Camera.interactive` is deprecated since 0.5.0 and will be removed in 0.6.0", category=FutureWarning)
    self.mouse_pan = value
    self.mouse_zoom = value

``` 

plus proper event emission. I do not remember how to do this. (@brisvag ?)

Reply:
I agree with @Czaki and @alisterburt that we should provide backwards compatibility as above. There's a "WarningEmitter" class (I thought you wrote it @Czaki??? 😂) for the deprecated events.

Reply:
> plus proper event emission. I do not remember how to do this.
It looks like the event emission of the dependent fields should be handled automatically:
https://github.com/napari/napari/blob/eca3919ce940b6ebab19552f4f2ab5d56fd0f74e/napari/utils/events/evented_model.py#L131-L170